packages/components/src/DateTimePickers/DateTimePicker/DateTimePicker.component.js

```diff
@@ -57,6 +87,20 @@ class DateTimePicker extends React.Component {
 		this.setState({ isDateTimeView });
 	}
 
+	trySubmit() {
+		if (this.state.selectedDate !== undefined &&
+			this.state.selectedTime !== undefined) {
+			this.submit();
+		}
+	}
+
+	submit() {
+		this.props.onSubmit({
```

Comment:
Why ?
I don't named it this way, because the **values can change when picked internally without be emitted outside**. They are emitted only when `submitted`/`validated` or an other word.
I understand submit can be misunderstood with form submit but if I set `onChange`, it will not be really trigger when the date or time changes :s.

Reply:
` when picked internally without be emitted outside`
But here we are talking about the `external` api. Form the outside world, it's a value change. Don't name exposed api things with internal concepts.

Reply:
The behavior of changing the picking value then submitted/validated it, is clear from the outside. It's the functional way the picker work, it's not technical.
=> I picked a date and a time, eventually change my picking multiple times, then I validate my choice (event without explicit user action it's a special case of validation)

If I use a input component with a 'done' button next to.
A `onChange`would be confusing : is it triggered when I change the input or when I submit it with the button or even both ?
A `onSubmit` is clear that it's not when the input change, but only when the button is used.

And by definition, all events/callbacks are emitted when a value change, it's not particularly inherent to `onChange`. There is just no need to precise for a single input because only text input can change, but for a complex widget, `onChange` is really confusing.

In the current `DateTimePicker` component context :
- `onChange` say to me "I will give you a DateTime"
- `onSubmit` say to me "I will give you a DateTime when I'm submitted"

Reply:
✘